### ISSUE TITLE
Fields: getRepeatable and attributes

### ISSUE THREAD

**Issue by Eighke** [akeeba/fof#422]
Nicholas,

I find a problem on the getRepeatable and the attributes. Currently most of the fields are using `$this->element['attributes']` to get the attribute value. No problem on `edit` or `item` form.

By it could be a problem on `browse` view if we want to change the value of the attribute on the fly. Because it will change the value for all the next row too. For example I have a bookstore component with a button to reserve. If the book is not avalaible, I want to empty the attributes `url` and add in `class` disabled.

```php
public function getRepeatable()
{
	if (!F0FModel::getTmpInstance('books', 'LibraryModel')->isAvalaible())
	{
		$this->element['url']   = "";
		$this->element['class'] = ' btn-inverse disabled';
	}

	return parent::getRepeatable();
}
```

It will work for my current row... but also for all the following row. So if the first row isn't avalaible it will disabled all my row~~~.

I see that in Joomla Field they are using `$this->attribute` instead, and init them in the `setup` for each instance of the field, so it solve the problem. Couldn't we use the same? Currently the only way to override the attributes for only the current row is:

```php
public function getRepeatable()
{
	if (!F0FModel::getTmpInstance('books', 'LibraryModel')->isAvalaible())
	{
		$url   = (string) $this->element['url'];
		$class = (string) $this->element['class'];

		$this->element['url']   = "";
		$this->element['class'] = ' btn-inverse disabled';

		$parent = parent::getRepeatable();

		$this->element['url']   = $url;
		$this->element['class'] = $class;

		return $parent;
	}

	return parent::getRepeatable();
}
```

**Comment by Eighke** [comment]
> If we do as you propose we add a major performance impact to the browse view which is the most used view type.

Uhm, I don't see any reason to have a major performance impact. Most of the `$this->attribute` are already created by JField on the `setup` (e.g. $this->class, $this->url). So they already are available to use.

So we can freely change :
```php
$class = $this->element['class'] ? ' class="' . (string) $this->element['class'] . '"' : '';
//To
$class = $this->class ? ' class="' . $this->class . '"' : '';
```
Without any performance drop.

------------------
------------------

After checking, it is available only for J! 3+, on 2.5 only some attributes are available ($this->required, $this->multiple, $this->description and $this->labelClass). So it is not possible if we need to keep the b/c with J! 2.5... Maybe keep the idea for FOF3.0 if you drop the compatibility with J! 2.5 ?

------------------

> Maybe you can get the regular URL and class in setup and store them as
object properties instead of hardcoding them in your field's code.

Yes, after some review I changed to that:

```php
public function setup(SimpleXMLElement $element, $value, $group = null)
{
        $this->url   = (string) $this->element['url'];
        $this->class = (string) $this->element['class'];

        return parent::setup($element, $value, $group);
}

public function getRepeatable()
{
    if (!F0FModel::getTmpInstance('books', 'LibraryModel')->isAvalaible())
    {
        $this->element['url']   = "";
        $this->element['class'] = ' btn-inverse disabled';

        $parent = parent::getRepeatable();

        $this->element['url']   = $this->url;
        $this->element['class'] = $this->class;

        return $parent;
    }

    return parent::getRepeatable();
}
```

**Comment by Eighke** [comment]
Roger that. :)
